### ISSUE TITLE
Log rotation and size limit

### ISSUE THREAD

**Issue by b4nst** [containerd/nerdctl#864]
Hello there, is there a way with `nerdctl` to configure a maximum log size and rotation akin what's done with the docker json log plugin?

**Comment by Zheaoli** [comment]
https://github.com/containerd/containerd/issues/4830

I'm not sure should nerdctl support this docker-daemon feature.

**Comment by fahedouch** [comment]
based on this https://github.com/containerd/containerd/issues/4830#issuecomment-744744375 containerd do not support log rotate. I think we can configure in `/etc/nerdctl/nerdctl.toml` how much logs the user should see! WDYT @AkihiroSuda  ?

**Comment by Zheaoli** [comment]
maybe we can add an argument in `tail -100 demo.log` style?

**Comment by fuweid** [comment]
containerd supports log plugin for runC shimv2. Maybe we can implement jsonfile plugin for that

**Comment by AkihiroSuda** [comment]
> maybe we can add an argument in tail -100 demo.log style?

No, the CLI syntax should emulate `docker run --log-opt XXX`.
https://docs.docker.com/config/containers/logging/configure/

> containerd supports log plugin for runC shimv2. Maybe we can implement jsonfile plugin for that

We have been already doing so: https://github.com/containerd/nerdctl/blob/v0.17.1/pkg/logging/logging.go

**Comment by fuweid** [comment]
> 
> > containerd supports log plugin for runC shimv2. Maybe we can implement jsonfile plugin for that
> 
> 
> 
> We have been already doing so: https://github.com/containerd/nerdctl/blob/v0.17.1/pkg/logging/logging.go
> 
> 
> 
Good to know this 👏. It seems that there is a way to get the customize parameters from MagicArg1's value, like max-size, max-files.

**Comment by fahedouch** [comment]
Yes @fuweid  this seems a good idea.  may be `nerdctl run --log-opt max-size=XX --log-opt max-file=XX` . I take this issue

**Comment by fahedouch** [comment]
fixed by https://github.com/containerd/nerdctl/pull/959. We only support json-file plugin for the moment.
